### PR TITLE
1.1 updates - Media Explorer library added for adding videos to posts

### PR DESCRIPTION
**Pull request by nimmolo**
See commit notes, please test!

### CODE REVIEW

File: external-videos.php [hunk 1]
```diff
@@ -47,17 +47,17 @@ public function __construct() {
...
-    require_once( plugin_dir_path( __FILE__ ) . 'core/ev-widget.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'core/ev-media-gallery.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'core/simple_html_dom.php' );
-
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-youtube.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-vimeo.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-dotsub.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-wistia.php' );
-    require_once( plugin_dir_path( __FILE__ ) . 'hosts/ev-dailymotion.php' );
+    require( plugin_dir_path( __FILE__ ) . 'core/ev-admin.php' );
```
Comment (silviapfeiffer) [inline external-videos.php:50]: Why did you replace the require_once includes?

File: hosts/dailymotion/mexp-ev-dailymotion-service.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+		// Create the response for the API
+		$response = new MEXP_Response();
+
+		if ( !isset( $query_response ) )
+			return false;
+
+    // Set up items
+		foreach ( $query_response as $index => $video ) {
+      setup_postdata( $video );
```
Comment (silviapfeiffer) [inline hosts/dailymotion/mexp-ev-dailymotion-service.php:72]: Indent seems a bit broken

File: hosts/youtube/mexp-ev-youtube-service.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    // error_log( '$query_response: ' . print_r( $query_response, true) );
+
+		// Create the response for the API
+		$response = new MEXP_Response();
+
+		if ( !isset( $query_response ) )
+			return false;
+
+    // Set up items
+		foreach ( $query_response as $index => $video ) {
```
Comment (silviapfeiffer) [inline hosts/youtube/mexp-ev-youtube-service.php]: more weird indents - sorry about being pedantic

**Review by silviapfeiffer — COMMENTED** [review]
Sorry, being a bit pedantic, but some of the code is inconsistently indented. I'll try it out now.

**Comment by silviapfeiffer** [comment]
Not your fault, but I'm getting:
`Warning: dirname() expects exactly 1 parameter, 2 given in .../wordpress/external-videos/mexp/media-explorer.php on line 20`

Note: I'm on php 5.6.16


You will want to replace 'dirname( __FILE__,2) ' with 'dirname(dirname( __FILE__)) ' - it's the easiest.

**Comment by silviapfeiffer** [comment]
BTW: I also noticed that https://wordpress.org/plugins/external-videos/screenshots/ isn't working. I think we need to add the directory `assets` to the readme file for the screenshots

File: external-videos.php (hunk 1 above)
Comment (nimmolo) [reply external-videos.php:50]: Researching the plugin i came across arguments that require_once required significantly more work from the server, and if the code has no chance of double includes, it's always better to require. Now that i'm looking it up again, seems it's not SUCH a big difference. http://stackoverflow.com/questions/186338/why-is-require-once-so-bad-to-use

File: hosts/dailymotion/mexp-ev-dailymotion-service.php (hunk 2 above)
Comment (nimmolo) [reply hosts/dailymotion/mexp-ev-dailymotion-service.php:72]: Yes this is weird! I have my editor set to do double spaces, no tabs, and everything seemed ok at commit time.

**Comment by nimmolo** [comment]
oh damn! didn't notice that dirname parameter was php 7-only.

File: hosts/youtube/mexp-ev-youtube-service.php (hunk 3 above)
Comment (nimmolo) [reply hosts/youtube/mexp-ev-youtube-service.php]: Fixed all these.

**Comment by nimmolo** [comment]
I looked through all the "Video" plugins currently on the WP repo, just to check out what's out there.

In light of that, and the fact that video embeds in WP are easier than ever, I'd like to propose a new tagline and description for what the plugin does, that more clearly differentiates it from others. Feel free to edit or discard.

File: external-videos.php (hunk 1 above)
Comment (silviapfeiffer) [reply external-videos.php:50]: Ah ok, I was just curious. Whatever works...

**Comment by silviapfeiffer** [comment]
I like the new tagline! 😀

**Comment by nimmolo** [comment]
Maybe we should add the word "sync" to the tags. The tags didnt update on SVN last time, did i do them right?

For the screenshots, I checked other plugins in SVN and it seems they shouldn't need the /assets directory folder listing, or even any filename at all, as long as they are sequentially numbered. 

Check here: https://plugins.svn.wordpress.org/wordpress-seo/trunk/readme.txt

I wonder if, on SVN, /assets maybe has to be synced separately, since it's outside of /trunk. Let me know if you want me to look into this further.

**Comment by silviapfeiffer** [comment]
Looking good. I'm landing and releasing it. Let's see if we get the tags updated.

**Comment by silviapfeiffer** [comment]
OK, released. The tags appeared, but they get ordered alphabetically.
Congratulations again! :-)

**Comment by nimmolo** [comment]
Thanks Silvia!

From [this newer page on the repo](https://developer.wordpress.org/plugins/wordpress-org/plugin-assets/) i'm reading the assets folder does have to be uploaded separately. I think that's why the icon and screenshots haven't been working. I'll see if i can fix it but i may not have direct access to SVN

**Comment by nimmolo** [comment]
Just checked - i don't have access to SVN as committer, but basically we just have to create an /assets top level folder separately from /branches, /tags, and /trunk.

screenshot

**Pull request merged**